src/main/java/com/mongodb/hibernate/internal/translate/AbstractMqlTranslator.java

```diff
@@ -379,22 +405,51 @@ public void visitNamedTableReference(NamedTableReference namedTableReference) {
 
     @Override
     public void visitRelationalPredicate(ComparisonPredicate comparisonPredicate) {
-        var astComparisonFilterOperator = getAstComparisonFilterOperator(comparisonPredicate.getOperator());
+        var fieldInLeftHandSide = visitFieldPath(comparisonPredicate.getLeftHandExpression());
+        var fieldInRightHandSide = visitFieldPath(comparisonPredicate.getRightHandExpression());
+
+        if (fieldInLeftHandSide == fieldInRightHandSide) {
+            if (fieldInLeftHandSide) {
+                throw new FeatureNotSupportedException(
+                        "MongoDB's comparison operation doesn't support comparison between two fields");
```

Comment:
Our translator doesn't currently support it, but MongoDB does, using `$expr`:

```
{$expr: {$eq: ["$x", "$y"]}}
```

So let's change the text to make it more accurate, as we may choose to implement this in the future.

Reply:
I changed it to:
```
Currently comparison between two fields not supported
```
and
```
Currently comparison between two values not supported
```